### ISSUE TITLE
[Bug]: Live feed seems to disconnect prematurely about half of the time (before stream can begin)

### ISSUE THREAD

**Issue by 6twenty** [Brandawg93/homebridge-nest-cam#430]
### Describe the bug

The live feed for my cameras are frequently failing with "No response" - though, every few attempts does seem to get a successful live feed streaming. I've ensured that my Mac Mini which runs homebridge only has one network interface/IP address active. The logs seem to indicate that the socket connection is closing prematurely, before ffmpeg has a chance to start delivering the video feed (note the "Stopped streaming session!") log in the middle of various ffmpeg logs).

### Debug Output

<details>
  <summary>Debug output when stream fails 🔽</summary>

```
[7/15/2021, 6:07:45 PM] [Nest-cam] VIDEO command: ffmpeg -f,h264,-use_wallclock_as_timestamps,1,-r,15,-i,pipe:,-c:v,h264_videotoolbox,-bf,0,-b:v,1196k,-bufsize,1196k,-maxrate,2392k,-pix_fmt,yuv420p,-an,-payload_type,99,-ssrc,14136543,-f,rtp,-srtp_out_suite,AES_CM_128_HMAC_SHA1_80,-srtp_out_params,XgWk5PdSQtoRkbA7pRZgCjNJMfwKc5VjcnU0TNiQ,srtp://192.168.178.84:54149?rtcpport=54149&localrtcpport=64160&pkt_size=1378
[7/15/2021, 6:07:45 PM] [Nest-cam] AUDIO command: ffmpeg -c:a,libfdk_aac,-i,pipe:,-c:a,libfdk_aac,-profile:a,aac_eld,-ac,1,-vn,-ar,16k,-b:a,24k,-flags,+global_header,-payload_type,110,-ssrc,16313112,-f,rtp,-srtp_out_suite,AES_CM_128_HMAC_SHA1_80,-srtp_out_params,dLrJ3WIMDgtROOXLLaNHX9oje3psC6GhagrY7kGm,srtp://192.168.178.84:53143?rtcpport=53143&localrtcpport=64161&pkt_size=188
[7/15/2021, 6:07:45 PM] [Nest-cam] RETURN AUDIO command: ffmpeg -hide_banner,-protocol_whitelist,pipe,udp,rtp,file,crypto,-f,sdp,-c:a,libfdk_aac,-i,pipe:0,-map,0:0,-c:a,libspeex,-frames_per_packet,4,-ac,1,-vn,-ar,16k,-f,data,pipe:1
[7/15/2021, 6:07:45 PM] [Nest-cam] Using HIGH quality stream.
[7/15/2021, 6:07:45 PM] [Nest-cam] waiting for socket to connect
[7/15/2021, 6:07:45 PM] [Nest-cam] VIDEO: ffmpeg version 4.4 Copyright (c) 2000-2021 the FFmpeg developers
  built with Apple clang version 11.0.0 (clang-1100.0.33.17)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/4.4-with-options_1 --enable-shared --cc=clang --host-cflags= --host-ldflags= --enable-gpl --enable-libaom --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-libsnappy --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-demuxer=dash --disable-libjack --disable-indev=jack --enable-opencl --enable-videotoolbox --disable-htmlpages --enable-libfdk-aac --enable-libspeex --enable-nonfree
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100

[7/15/2021, 6:07:45 PM] [Nest-cam] AUDIO: ffmpeg version 4.4
[7/15/2021, 6:07:45 PM] [Nest-cam] AUDIO:  Copyright (c) 2000-2021 the FFmpeg developers
  built with Apple clang version 11.0.0 (clang-1100.0.33.17)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/4.4-with-options_1 --enable-shared --cc=clang --host-cflags= --host-ldflags= --enable-gpl --enable-libaom --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-libsnappy --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-demuxer=dash --disable-libjack --disable-indev=jack --enable-opencl --enable-videotoolbox --disable-htmlpages --enable-libfdk-aac --enable-libspeex --enable-nonfree
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100

[7/15/2021, 6:07:46 PM] [Nest-cam] [NexusStreamer] Connected
[7/15/2021, 6:07:47 PM] [Nest-cam] [NexusStreamer] OK
[7/15/2021, 6:07:47 PM] [Nest-cam] [NexusStreamer] Redirect
[7/15/2021, 6:07:47 PM] [Nest-cam] [NexusStreamer] Redirecting...
[7/15/2021, 6:07:48 PM] [Nest-cam] [NexusStreamer] Disconnected
[7/15/2021, 6:07:48 PM] [Nest-cam] Using HIGH quality stream.
[7/15/2021, 6:07:48 PM] [Nest-cam] waiting for socket to connect
[7/15/2021, 6:07:49 PM] [Nest-cam] [NexusStreamer] Connected
[7/15/2021, 6:07:50 PM] [Nest-cam] [NexusStreamer] OK
[7/15/2021, 6:07:52 PM] [Nest-cam] [NexusStreamer] Playback Begin
[7/15/2021, 6:07:53 PM] [Nest-cam] AUDIO: Input #0, aac, from 'pipe:':
  Duration: N/A, bitrate: 65 kb/s
  Stream #0:0: Audio: aac, 48000 Hz, mono, s16, 65 kb/s

[7/15/2021, 6:07:53 PM] [Nest-cam] AUDIO: Stream mapping:
  Stream #0:0 -> #0:0 (aac (libfdk_aac) -> aac (libfdk_aac))

[7/15/2021, 6:07:53 PM] [Nest-cam] AUDIO: Output #0, rtp, to 'srtp://192.168.178.84:53143?rtcpport=53143&localrtcpport=64161&pkt_size=188':
  Metadata:
    encoder         : Lavf58.76.100
  Stream #0:0: Audio: aac (ELD), 16000 Hz, mono, s16, 24 kb/s
    Metadata:
      encoder         : Lavc58.134.100 libfdk_aac
size=       0kB time=00:00:00.00 bitrate=N/A speed=   0x
[7/15/2021, 6:07:54 PM] [Nest-cam] AUDIO: size=       8kB time=00:00:02.22 bitrate=  29.3kbits/s speed=2.83x
[7/15/2021, 6:07:54 PM] [Nest-cam] Stopped streaming session!
[7/15/2021, 6:07:54 PM] [Nest-cam] RETURN AUDIO: Guessed Channel Layout for Input Stream #0.0 : mono
Input #0, sdp, from 'pipe:0':
  Metadata:
    title           : Talk
  Duration: N/A, bitrate: N/A
  Stream #0:0: Audio: aac, 16000 Hz, mono, s16

[7/15/2021, 6:07:54 PM] [Nest-cam] RETURN AUDIO: Stream mapping:
  Stream #0:0 -> #0:0 (aac (libfdk_aac) -> speex (libspeex))
Finishing stream 0:0 without any data written to it.

[7/15/2021, 6:07:54 PM] [Nest-cam] AUDIO: size=      11kB time=00:00:03.02 bitrate=  29.0kbits/s speed= 2.5x
video:0kB audio:9kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 16.871330%

[7/15/2021, 6:07:54 PM] [Nest-cam] AUDIO: Exiting normally, received signal 15.

[7/15/2021, 6:07:54 PM] [Nest-cam] VIDEO: Input #0, h264, from 'pipe:':
  Duration: N/A, start: 1626329272.244336, bitrate: N/A
  Stream #0:0: Video: h264 (Main), yuvj420p(pc, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 7.50 fps, 7.50 tbr, 1200k tbn, 15 tbc

[7/15/2021, 6:07:54 PM] [Nest-cam] RETURN AUDIO: Output #0, data, to 'pipe:1':
  Metadata:
    title           : Talk
    encoder         : Lavf58.76.100

[7/15/2021, 6:07:54 PM] [Nest-cam] RETURN AUDIO:   Stream #0:0: Audio: speex, 16000 Hz, mono, s16, 27 kb/s
    Metadata:
      encoder         : Lavc58.134.100 libspeex
size=       0kB time=00:00:00.00 bitrate=N/A speed=   0x
video:0kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown

[7/15/2021, 6:07:54 PM] [Nest-cam] RETURN AUDIO: Exiting normally, received signal 15.

[7/15/2021, 6:07:54 PM] [Nest-cam] VIDEO: Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> h264 (h264_videotoolbox))

[7/15/2021, 6:07:54 PM] [Nest-cam] VIDEO: Finishing stream 0:0 without any data written to it.

[7/15/2021, 6:07:54 PM] [Nest-cam] VIDEO: [swscaler @ 0x7f8dbb136000] deprecated pixel format used, make sure you did set range correctly

[7/15/2021, 6:07:54 PM] [Nest-cam] VIDEO: Output #0, rtp, to 'srtp://192.168.178.84:54149?rtcpport=54149&localrtcpport=64160&pkt_size=1378':
  Metadata:
    encoder         : Lavf58.76.100
  Stream #0:0: Video: h264, yuv420p, 1920x1080 [SAR 1:1 DAR 16:9], q=2-31, 1196 kb/s, 15 fps, 90k tbn
[7/15/2021, 6:07:54 PM] [Nest-cam] VIDEO:
    Metadata:
      encoder         : Lavc58.134.100 h264_videotoolbox

[7/15/2021, 6:07:54 PM] [Nest-cam] VIDEO: frame=    0 fps=0.0 q=0.0 Lsize=       0kB time=00:00:00.00 bitrate=N/A speed=   0x

[7/15/2021, 6:07:54 PM] [Nest-cam] VIDEO: video:0kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
Output file is empty, nothing was encoded (check -ss / -t / -frames parameters if used)

[7/15/2021, 6:07:54 PM] [Nest-cam] VIDEO: Exiting normally, received signal 15.

[7/15/2021, 6:07:54 PM] [Nest-cam] Stopped streaming session!
[7/15/2021, 6:07:55 PM] [Nest-cam] [NexusStreamer] Playback End
[7/15/2021, 6:07:55 PM] [Nest-cam] [NexusStreamer] Disconnected
```

</details>

<details>
  <summary>Debug output when stream succeeds 🔽</summary>

```
[7/15/2021, 6:23:13 PM] [Nest-cam] VIDEO command: ffmpeg -f,h264,-use_wallclock_as_timestamps,1,-r,15,-i,pipe:,-c:v,h264_videotoolbox,-bf,0,-b:v,1196k,-bufsize,1196k,-maxrate,2392k,-pix_fmt,yuv420p,-an,-payload_type,99,-ssrc,16137194,-f,rtp,-srtp_out_suite,AES_CM_128_HMAC_SHA1_80,-srtp_out_params,wggEi/VSgHeoa8f6VNKSnuyhcqvs9/JjEPwQpKoE,srtp://192.168.178.77:62017?rtcpport=62017&localrtcpport=50710&pkt_size=1378
[7/15/2021, 6:23:13 PM] [Nest-cam] AUDIO command: ffmpeg -c:a,libfdk_aac,-i,pipe:,-c:a,libfdk_aac,-profile:a,aac_eld,-ac,1,-vn,-ar,16k,-b:a,24k,-flags,+global_header,-payload_type,110,-ssrc,3318442,-f,rtp,-srtp_out_suite,AES_CM_128_HMAC_SHA1_80,-srtp_out_params,Hh/pwjVxHxit7kzGtdXb80dMCtr1A5jjHj9jbUZy,srtp://192.168.178.77:59672?rtcpport=59672&localrtcpport=50711&pkt_size=188
[7/15/2021, 6:23:13 PM] [Nest-cam] RETURN AUDIO command: ffmpeg -hide_banner,-protocol_whitelist,pipe,udp,rtp,file,crypto,-f,sdp,-c:a,libfdk_aac,-i,pipe:0,-map,0:0,-c:a,libspeex,-frames_per_packet,4,-ac,1,-vn,-ar,16k,-f,data,pipe:1
[7/15/2021, 6:23:13 PM] [Nest-cam] Using HIGH quality stream.
[7/15/2021, 6:23:13 PM] [Nest-cam] waiting for socket to connect
[7/15/2021, 6:23:13 PM] [Nest-cam] VIDEO: ffmpeg version 4.4 Copyright (c) 2000-2021 the FFmpeg developers

[7/15/2021, 6:23:13 PM] [Nest-cam] VIDEO:   built with Apple clang version 11.0.0 (clang-1100.0.33.17)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/4.4-with-options_1 --enable-shared --cc=clang --host-cflags= --host-ldflags= --enable-gpl --enable-libaom --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-libsnappy --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-demuxer=dash --disable-libjack --disable-indev=jack --enable-opencl --enable-videotoolbox --disable-htmlpages --enable-libfdk-aac --enable-libspeex --enable-nonfree
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100

[7/15/2021, 6:23:13 PM] [Nest-cam] AUDIO: ffmpeg version 4.4 Copyright (c) 2000-2021 the FFmpeg developers
[7/15/2021, 6:23:13 PM] [Nest-cam] AUDIO:
  built with Apple clang version 11.0.0 (clang-1100.0.33.17)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/4.4-with-options_1 --enable-shared --cc=clang --host-cflags= --host-ldflags= --enable-gpl --enable-libaom --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-libsnappy --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-demuxer=dash --disable-libjack --disable-indev=jack --enable-opencl --enable-videotoolbox --disable-htmlpages --enable-libfdk-aac --enable-libspeex --enable-nonfree
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100

[7/15/2021, 6:23:14 PM] [Nest-cam] [NexusStreamer] Connected
[7/15/2021, 6:23:14 PM] [Nest-cam] [NexusStreamer] OK
[7/15/2021, 6:23:15 PM] [Nest-cam] [NexusStreamer] Redirect
[7/15/2021, 6:23:15 PM] [Nest-cam] [NexusStreamer] Redirecting...
[7/15/2021, 6:23:15 PM] [Nest-cam] [NexusStreamer] Playback End
[7/15/2021, 6:23:15 PM] [Nest-cam] [NexusStreamer] Disconnected
[7/15/2021, 6:23:15 PM] [Nest-cam] Using HIGH quality stream.
[7/15/2021, 6:23:15 PM] [Nest-cam] waiting for socket to connect
[7/15/2021, 6:23:16 PM] [Nest-cam] [NexusStreamer] Connected
[7/15/2021, 6:23:17 PM] [Nest-cam] [NexusStreamer] OK
[7/15/2021, 6:23:18 PM] [Nest-cam] [NexusStreamer] Playback Begin
[7/15/2021, 6:23:19 PM] [Nest-cam] AUDIO: Input #0, aac, from 'pipe:':
  Duration: N/A, bitrate: 65 kb/s
  Stream #0:0: Audio: aac, 48000 Hz, mono, s16, 65 kb/s

[7/15/2021, 6:23:19 PM] [Nest-cam] AUDIO: Stream mapping:
  Stream #0:0 -> #0:0
[7/15/2021, 6:23:19 PM] [Nest-cam] AUDIO:  (aac (libfdk_aac) -> aac (libfdk_aac))

[7/15/2021, 6:23:19 PM] [Nest-cam] AUDIO: Output #0, rtp, to 'srtp://192.168.178.77:59672?rtcpport=59672&localrtcpport=50711&pkt_size=188':

[7/15/2021, 6:23:19 PM] [Nest-cam] AUDIO:   Metadata:
    encoder         : Lavf58.76.100
  Stream #0:0: Audio: aac (ELD), 16000 Hz, mono, s16, 24 kb/s
    Metadata:
      encoder         : Lavc58.134.100 libfdk_aac

[7/15/2021, 6:23:19 PM] [Nest-cam] AUDIO: size=       0kB time=00:00:00.00 bitrate=N/A speed=   0x
[7/15/2021, 6:23:20 PM] [Nest-cam] AUDIO: size=       6kB time=00:00:01.77 bitrate=  29.6kbits/s speed=3.04x
[7/15/2021, 6:23:21 PM] [Nest-cam] AUDIO: size=       8kB time=00:00:02.19 bitrate=  29.4kbits/s speed=1.88x
[7/15/2021, 6:23:21 PM] [Nest-cam] VIDEO: Input #0, h264, from 'pipe:':
  Duration: N/A, start: 1626330198.891953, bitrate: N/A

[7/15/2021, 6:23:21 PM] [Nest-cam] VIDEO:   Stream #0:0: Video: h264 (Main), yuvj420p(pc, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 15 fps, 15 tbr, 1200k tbn, 30 tbc

[7/15/2021, 6:23:21 PM] [Nest-cam] VIDEO: Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> h264 (h264_videotoolbox))

[7/15/2021, 6:23:21 PM] [Nest-cam] VIDEO: [swscaler @ 0x7fb807f47000] deprecated pixel format used, make sure you did set range correctly

[7/15/2021, 6:23:21 PM] [Nest-cam] VIDEO: Output #0, rtp, to 'srtp://192.168.178.77:62017?rtcpport=62017&localrtcpport=50710&pkt_size=1378':
  Metadata:
    encoder         : Lavf58.76.100

[7/15/2021, 6:23:21 PM] [Nest-cam] VIDEO:   Stream #0:0: Video: h264, yuv420p(tv, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], q=2-31, 1196 kb/s, 15 fps, 90k tbn
    Metadata:
      encoder         : Lavc58.134.100 h264_videotoolbox

[7/15/2021, 6:23:21 PM] [Nest-cam] VIDEO: frame=    1 fps=0.0 q=0.0 size=       0kB time=00:00:00.00 bitrate=N/A speed=   0x
[7/15/2021, 6:23:21 PM] [Nest-cam] AUDIO: size=      10kB time=00:00:02.79 bitrate=  29.1kbits/s speed=1.51x
[7/15/2021, 6:23:21 PM] [Nest-cam] VIDEO: frame=   38 fps=0.0 q=-0.0 size=     264kB time=00:00:01.93 bitrate=1118.6kbits/s speed=2.78x
```
</details>

### Steps to reproduce

1. In iOS Home app, tap a camera tile to start the live feed
2. After a few seconds, either the stream begins or it shows "No response. This camera is not responding"

### Device Type

Mac Mini

### iOS Version

14.6

### Camera Types

Nest Cam Outdoor (x2)

### Bug Persistence

Randomly

### Last Working Version

N/A - first time setting up homebridge-nest-cam

**Comment by Brandawg93** [comment]
Try ensuring that `Network Interfaces` is set to your current interface and if that doesn't work, try changing your `mDNS Advertiser` to a different option. All of this is under Homebridge Settings in the UI.

**Comment by 6twenty** [comment]
Thanks! Will try these out today

**Comment by 6twenty** [comment]
Update: I've set the network interface to `en0`, which is the only active interface (wifi is off on this device) and tried changing the mDNS advertiser to "ciao", under which I wasn't able to get the camera stream to start at all. Switching back to "bonjour-hap" for the stream working for a little while, but it has since gone back to the same original behaviour - failing to start about 50% of the time.

For what it's worth, these are the bridge config settings I ended up with:

```
  "bridge": {
    "name": "Homebridge",
    "username": "CC:22:3D:E3:CE:45",
    "port": 51826,
    "pin": "031-45-169",
    "advertiser": "bonjour-hap",
    "bind": ["en0"]
  }
```


The "advertiser" and "bind" options are the ones I just added (using https://github.com/homebridge/homebridge/wiki/mDNS-Options as my reference).
